### PR TITLE
Add index name validation

### PR DESCRIPTION
**Pull request by VijayanB**
### Description
index name should end with suffix -map.

Signed-off-by: Vijayan Balasubramanian <balasvij@amazon.com>

 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/geospatial/action/upload/geojson/UploadGeoJSONRequestContent.java [hunk 1]
```diff
@@ -69,6 +68,24 @@ public static UploadGeoJSONRequestContent create(Map<String, Object> input) {
         return new UploadGeoJSONRequestContent(index, fieldName, fieldType, (List<Object>) geoJSONData);
     }
 
+    private static String validateIndexName(Map<String, Object> input) {
+        String index = extractValueAsString(input, FIELD_INDEX.getPreferredName());
+        if (!Strings.hasText(index)) {
+            throw new IllegalArgumentException("field [ " + FIELD_INDEX.getPreferredName() + " ] cannot be empty");
```
Comment (martin-gaievski) [inline src/main/java/org/opensearch/geospatial/action/upload/geojson/UploadGeoJSONRequestContent.java:74]: can we use String.format here?

File: src/test/java/org/opensearch/geospatial/action/upload/geojson/UploadGeoJSONRequestContentTests.java [hunk 2]
```diff
@@ -33,7 +36,8 @@ private Map<String, Object> buildRequestContent(String indexName, String fieldNa
     }
 
     public void testCreate() {
-        final String indexName = "some-index";
+        final String indexName = randomLowerCaseStringWithSuffix(ACCEPTED_INDEX_SUFFIX_PATH);
+        ;
```
Comment (martin-gaievski) [inline src/test/java/org/opensearch/geospatial/action/upload/geojson/UploadGeoJSONRequestContentTests.java:40]: probably we don't need it here

File: src/test/java/org/opensearch/geospatial/rest/action/upload/geojson/RestUploadGeoJSONActionIT.java [hunk 3]
```diff
@@ -54,7 +57,8 @@ public void testGeoJSONUploadFailIndexExists() throws IOException {
 
     public void testGeoJSONUploadSuccessPutMethod() throws IOException {
 
-        String index = randomLowerCaseString();
+        String index = randomLowerCaseStringWithSuffix(ACCEPTED_INDEX_SUFFIX_PATH);
+        ;
```
Comment (martin-gaievski) [inline src/test/java/org/opensearch/geospatial/rest/action/upload/geojson/RestUploadGeoJSONActionIT.java:61]: same, we don't need this single ";" here

File: src/main/java/org/opensearch/geospatial/action/upload/geojson/UploadGeoJSONRequestContent.java [hunk 4]
```diff
@@ -45,10 +47,7 @@ private UploadGeoJSONRequestContent(String indexName, String fieldName, String f
      */
     public static UploadGeoJSONRequestContent create(Map<String, Object> input) {
         Objects.requireNonNull(input, "input cannot be null");
-        String index = extractValueAsString(input, FIELD_INDEX.getPreferredName());
-        if (!Strings.hasText(index)) {
-            throw new IllegalArgumentException("field [ " + FIELD_INDEX.getPreferredName() + " ] cannot be empty");
-        }
+        String index = validateIndexName(input);
```
Comment (Shivamdhar) [inline src/main/java/org/opensearch/geospatial/action/upload/geojson/UploadGeoJSONRequestContent.java:50]: [nit] we can make index final.

File: src/test/java/org/opensearch/geospatial/rest/action/upload/geojson/RestUploadGeoJSONActionIT.java (hunk 3 above)
Comment (VijayanB) [reply src/test/java/org/opensearch/geospatial/rest/action/upload/geojson/RestUploadGeoJSONActionIT.java]: Ack

File: src/main/java/org/opensearch/geospatial/action/upload/geojson/UploadGeoJSONRequestContent.java (hunk 4 above)
Comment (VijayanB) [reply src/main/java/org/opensearch/geospatial/action/upload/geojson/UploadGeoJSONRequestContent.java]: Ack

File: src/main/java/org/opensearch/geospatial/action/upload/geojson/UploadGeoJSONRequestContent.java (hunk 1 above)
Comment (VijayanB) [reply src/main/java/org/opensearch/geospatial/action/upload/geojson/UploadGeoJSONRequestContent.java]: Ack

**Review by Shivamdhar — APPROVED** [review]

**Review by martin-gaievski — APPROVED** [review]
looks good, thank you

**Pull request merged**